qtfred/src/mission/dialogs/ReinforcementsEditorDialogModel.cpp

```diff
@@ -0,0 +1,324 @@
+#include "ReinforcementsEditorDialogModel.h"
+#include "ship/ship.h"
+#include <algorithm>
+
+namespace fso {
+namespace fred {
+namespace dialogs {
+
+ReinforcementsDialogModel::ReinforcementsDialogModel(QObject* parent, EditorViewport* viewport)
+	: AbstractDialogModel(parent, viewport)
+{
+	initializeData();
+}
+
+void ReinforcementsDialogModel::initializeData()
+{
+	for (int i = 0; i < Num_reinforcements; i++) {
+		_reinforcementList.emplace_back(Reinforcements[i].name, Reinforcements[i].uses, Reinforcements[i].arrival_delay);
+	}
+
+	// add the wings to the model's internal storage
+	for (auto& currentWing : Wings) {
+
+		if (currentWing.wave_count == 0) {
+			continue;
+		}
+
+		// no players allowed in reinforcements!
+		for (auto& playerTest : currentWing.ship_index) {
+			if (playerTest > -1 && (Objects[Ships[playerTest].objnum].type == OBJ_START)){
+				continue;
```

Comment:
You're going to run into the problem I mentioned earlier where `continue` only continues the innermost loop.

Reply:
Ok, test bool added.